sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_internal/async_challenge_auth_policy.py

```diff
@@ -67,8 +59,31 @@ async def on_challenge(self, request: "PipelineRequest", response: "PipelineResp
 
         body = request.context.pop("key_vault_request_data", None)
         request.http_request.set_text_body(body)  # no-op when text is None
-
-        self._last_tenant_id = challenge.tenant_id
         await self.authorize_request(request, *self._scopes, tenant_id=challenge.tenant_id)
 
         return True
+
+    async def _handle_challenge(self, request: "PipelineRequest", challenge: "HttpChallenge") -> None:
+        """Authenticate according to challenge, add Authorization header to request"""
+
+        scope = challenge.get_scope() or challenge.get_resource() + "/.default"
+        await self.authorize_request(request, scope, tenant_id=challenge.tenant_id)
+
+    async def _handle_response(self, request: "PipelineRequest", response: "PipelineResponse") -> "PipelineResponse":
```

Comment:
What went wrong with `on_challenge`?

Reply:
Adding `_handle_response` was actually a mistake -- I can't really say why I thought this was necessary. It's also responsible for all the playback errors, as I'm sure you noticed if you looked at the pipeline. My bad!